app/javascript/packages/download-button/download-button-element.ts

```diff
@@ -0,0 +1,56 @@
+declare global {
+  interface Navigator {
+    msSaveBlob?: (blob: Blob, filename: string) => void;
+  }
+}
+
+/**
+ * Converts a data URI to a Blob. The given URI data must be URI-encoded and have a MIME type of
+ * `text/plain`.
+ *
+ * @param uri URI string to convert.
+ *
+ * @return Blob instance.
+ */
+export function dataURIToBlob(uri: string) {
+  const data = decodeURIComponent(uri.split(',')[1]);
+  const bytes = Uint8Array.from(data.split('').map((char) => char.charCodeAt(0)));
```

Comment:
Can we pass the `mapFn` to the constructor?
```suggestion
  const bytes = Uint8Array.from(data, (char) => char.charCodeAt(0));
```

Reply:
> Can we pass the `mapFn` to the constructor?

Good call!

Reply:
Updated in rebased dc8ca43

Reply:
FWIW, I was copying in part from existing code, so we may consider making a similar simplification there.

https://github.com/18F/identity-idp/blob/3e90edd8331087e985872f9031c5bd2ab49dbc6a/app/javascript/app/webauthn.js#L1-L9